src/figuresClasses.ts

```diff
@@ -1,19 +1,104 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape;
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.shape = 'triangle';
+    this.isValidParameters();
+  }
+
+  isValidParameters(): void {
+    if (
+      this.a <= 0
+      || this.b <= 0
+      || this.c <= 0
+    ) {
+      throw new Error('side(s) can\'t be less then 0');
+    }
```

Comment:
it seems like you can do these operations inside a constructor because every class has a different implementation of this function